src/app/config/config.c

```diff
@@ -4821,11 +4817,14 @@ opt_streq(const char *s1, const char *s2)
 
 /** Check if any of the previous options have changed but aren't allowed to. */
 static int
-options_transition_allowed(const or_options_t *old,
-                           const or_options_t *new_val,
-                           char **msg)
+options_check_transition_cb(const void *old_,
+                            const void *new_val_,
+                            char **msg)
 {
-  if (!old)
+  const or_options_t *old = old_;
+  const or_options_t *new_val = new_val_;
```

Comment:
Should we check the magic values before this cast?

Do we need to change any other functions so they check magic values, rather than just casting?